src/autoscaling/snapshotter.js

```diff
@@ -360,6 +430,23 @@ class Snapshotter {
         }
         snapshots.splice(0, oldCount);
     }
+
+    /**
+     * Calculate max memory for platform or local usage.
+     * @ignore
+     */
+    async _ensureCorrectMaxMemory() {
+        if (!this.maxMemoryBytes) {
+            const { totalBytes } = await getMemoryInfo();
```

Comment:
On the platform, I'd skip the call to `getMemoryInfo()` altogether, the spawning of the process is slow and you don't really need it just to get `totalBytes`

Reply:
Actually, an even better way would be for `getMemoryInfo` to use data from `systemInfo` event on the platform and `ps` on local. This way, you can call it without any performance penalty.

Reply:
But then you'd need to collect the `systemInfo` events somewhere to be able to serve them on demand with `getMemoryInfo`.

It does not call the function on the platform because `this.maxMemoryBytes` should always be set on platform (in constructor, using `APIFY_MEMORY_MBYTES`). It would only call the function if it were not set for some reason.

Reply:
Oh, you're right, I missed the condition on top. BTW, it's much better to keep code to the left side as much as possible, see the "Keep code left" paradigm, because then it's easier to read. in this case to immediately return if `this.maxMemoryBytes` is set.

Anyway, I think `getMemoryInfo()` on platform should use the the `systemInfo` message, rather than rely on `ps`. But that's for another time...